crates/runc-rust/src/specs.rs

```diff
@@ -0,0 +1,380 @@
+/*
+   copyright the containerd authors.
+
+   licensed under the apache license, version 2.0 (the "license");
+   you may not use this file except in compliance with the license.
+   you may obtain a copy of the license at
+
+       http://www.apache.org/licenses/license-2.0
+
+   unless required by applicable law or agreed to in writing, software
+   distributed under the license is distributed on an "as is" basis,
+   without warranties or conditions of any kind, either express or implied.
+   see the license for the specific language governing permissions and
+   limitations under the license.
+*/
+
+// Forked from https://github.com/pwFoo/rust-runc/blob/master/src/specs.rs
+/*
+ * Copyright 2020 fsyncd, Berlin, Germany.
+ * Additional material, copyright of the containerd authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+/// The dimensions of a rectangle, used for console sizing
+/// This name is different from Go Version, avoiding confilict with [`std::boxed::Box`]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BoxSize {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub height: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub width: Option<usize>,
+}
+
+/// User and group information for the container's process
+#[derive(Debug, Serialize, Deserialize)]
+pub struct User {
```

Comment:
I think we can replace the whole `specs.rs` with `oci-spec` crate: https://github.com/containers/oci-spec-rs

Reply:
I was not aware of that crate, thank you!
However, it seems to conflict because `oci-spec` requires rustc 1.54+ but `rust-toolchain.toml` in this repository specifies rustc 1.52 ?

Reply:
I think its fine to update MSRV to 1.54

Reply:
Ok, thank you!